### PR TITLE
Issue #SB-15780 feat: Added API Integration in library page

### PR DESCRIPTION
**Pull request by itsvick**

### CODE REVIEW

**Comment by itsvick** [comment]
@harishgilimi 
**Dependency**: @project-sunbird/common-consumption needs to be published.

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 1]
```diff
@@ -7,16 +7,17 @@ import * as _ from 'lodash-es';
...
     selector: 'app-library-filters',
     templateUrl: './library-filters.component.html',
     styleUrls: ['./library-filters.component.scss']
 })
-export class LibraryFiltersComponent implements OnInit {
+export class LibraryFiltersComponent implements OnInit, OnChanges {
     frameworkName: string;
     hashTagId: string;
-
+    selectedBoard: string;
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: which property holds selected medium and class

Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: in the html i can see state,  here it is board. can you confirm on this

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 2]
```diff
@@ -47,6 +52,10 @@ export class LibraryFiltersComponent implements OnInit {
             });
     }
 
+    ngOnChanges(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: remove this method

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 3]
```diff
@@ -61,9 +70,26 @@ export class LibraryFiltersComponent implements OnInit {
                     break;
             }
         });
+
+        this.libraryFilters.emit([]);
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: this should emit all selected filter

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 4]
```diff
@@ -61,9 +70,26 @@ export class LibraryFiltersComponent implements OnInit {
...
     }
 
     getSelectedFilters(event, type) {
+        this.filterQueryParam.board = ['CBSE'];
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: shoudnt be hardcoded

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 5]
```diff
@@ -61,9 +70,26 @@ export class LibraryFiltersComponent implements OnInit {
...
+
+        if (type === 'medium') {
+            this.filterQueryParam.medium = [event.data.text];
+        } else if (type === 'class') {
+            this.filterQueryParam.gradeLevel = [event.data.text];
+        }
+
+        this.filterQueryParam['appliedFilters'] = true;
+        this.router.navigate([], { relativeTo: this.activatedRoute.parent, queryParams: this.filterQueryParam });
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: is this component emits event when filter changes or routes to fetch relevant content

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 6]
```diff
@@ -61,9 +70,26 @@ export class LibraryFiltersComponent implements OnInit {
...
         console.log('Event', event);
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: remove consoles

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 7]
```diff
@@ -61,9 +70,26 @@ export class LibraryFiltersComponent implements OnInit {
...
+        console.log('filterQueryParam', this.filterQueryParam);
+    }
+
+    applyFilters() {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: remove this method if not used

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 8]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
+        this.setNoResultMessage();
     }
 
     ngOnDestroy() {
         this.unsubscribe$.next();
         this.unsubscribe$.complete();
     }
 
+
+    getFilters(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: we dont need this method, default filter should be taken from user preferance

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 9]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
+        private pageApiService: PageApiService,
+        private utilService: UtilService,
+        private toasterService: ToasterService,
+        private configService: ConfigService,
+        private resourceService: ResourceService,
+        private orgDetailsService: OrgDetailsService
     ) { }
 
     ngOnInit() {
+        this.setNoResultMessage();
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: resource service will get data by making api call. If the data has not fetched when this method is called all properties will be set as undefined or we might get error. And also this method is not considering language change. Pass this data to no result component in html it self

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 10]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
         this.unsubscribe$.complete();
     }
 
+
+    getFilters(event) {
+        this.dataDrivenFilters = {
+            board: 'CBSE'
+        };
+
+        this.fetchContentOnParamChange();
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: call this method on ngOninit

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 11]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
+        const filters = _.pickBy(this.queryParams, (value: Array<string> | string, key) => {
+            if (_.includes(['sort_by', 'sortType', 'appliedFilters'], key)) {
+                return false;
+            }
+            return value.length;
+        });
+        const softConstraintData = {
+            filters: {
+                channel: this.hashTagId,
+                board: [this.dataDrivenFilters.board]
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: should be taken from user preferance

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 12]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
+            };
+        } else {
+            this.noResultMessage = {
+                message: 'messages.stmsg.m0007',
+                messageText: 'messages.stmsg.m0006'
+            };
+        }
+    }
+
     onViewAllClick(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: this needs to be handled

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 13]
```diff
@@ -50,4 +176,8 @@ export class LibraryComponent implements OnInit {
     openContent(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: this needs to be handled

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 14]
```diff
@@ -50,4 +176,8 @@ export class LibraryComponent implements OnInit {
...
         console.log('Event', event);
     }
+
+    afterChange(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: what is this method for

File: src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts [hunk 15]
```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
...
+                messageText: 'messages.stmsg.m0006'
+            };
+        }
+    }
+
     onViewAllClick(event) {
-        console.log('Event', event);
     }
 
     showAllList(event) {
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts]: what is this method for

File: src/app/client/projects/desktop/src/app/modules/offline/components/desktop-explore-content/desktop-explore-content.component.ts [hunk 16]
```diff
@@ -0,0 +1,311 @@
+import {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/desktop-explore-content/desktop-explore-content.component.ts]: pagination is not there for explore content in my library

File: src/app/client/projects/desktop/src/app/modules/offline/components/desktop-view-all/desktop-view-all.component.html [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+        </div>
+    </div>
+    <div class="twelve wide column" *ngIf="noResult && !showLoader">
+        <app-no-result [data]="noResultMessage"></app-no-result>
+    </div>
+    <div class="twelve wide column" *ngIf="showLoader">
+        <app-loader [data]='loaderMessage'></app-loader>
+    </div>
+
+    <div class="twelve wide column right aligned py-0"
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/desktop-view-all/desktop-view-all.component.html]: pagination not there here also for view all in my library

File: src/app/client/projects/desktop/src/app/modules/offline/components/desktop-view-all/desktop-view-all.component.ts [hunk 18]
```diff
@@ -0,0 +1,384 @@
...
+        private userService: UserService,
+        private browserCacheTtlService: BrowserCacheTtlService,
+    ) {
+        this.router.onSameUrlNavigation = 'reload';
+        this.sortingOptions = this.configService.dropDownConfig.FILTER.RESOURCES.sortingOptions;
+    }
+
+    ngOnInit() {
+        if (!this.userService.loggedIn) {
+            this.getChannelId();
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/desktop-view-all/desktop-view-all.component.ts]: we don't have logged in functionality

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 19]
```diff
@@ -1,69 +1,167 @@
...
-                return this.frameworkService.frameworkData$;
-            }), first()).subscribe((data) => {
-                if (data && data.frameworkdata && data.frameworkdata.defaultFramework && data.frameworkdata.defaultFramework.categories) {
-                    this.frameworkCategories = data.frameworkdata.defaultFramework.categories;
-                    this.setFilters();
+        this.userDetails = this.onboardingService.userData;
+        this.setBMC();
+    }
+
+    setBMC() {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: This method should be meaningful

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 20]
```diff
@@ -1,69 +1,167 @@
...
+            this.selectedMediumIndex = [event.data.index];
+            this.selectedClassIndex = [];
+        } else if (type === 'class') {
+            this.selectedClassIndex = [event.data.index];
+        }
+
+        this.filterChange.emit(this.getSelectedBMCData());
+    }
+
+    getSelectedBMCData() {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: This method name also needs to update

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 21]
```diff
@@ -1,69 +1,186 @@
...
+        });
+    }
+
+    setBoard() {
+        this.channelService.getFrameWork(this.hashTagId).subscribe(orgDetails => {
+            this.boards = _.get(orgDetails, 'result.channel.frameworks');
+
+            if (this.boards) {
+
+                if (this.selectedFilters && this.selectedFilters.board && this.selectedFilters.board[0]) {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: use _.get here like _.get(this, 'selectedFilters.board[0]')

File: src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts [hunk 22]
```diff
@@ -1,69 +1,186 @@
...
         this.frameworkCategories.forEach(element => {
             switch (element.code) {
-                case 'board':
-                    this.boards = element.terms.map(state => state.name);
-                    break;
                 case 'medium':
                     this.mediums = element.terms.map(medium => medium.name);
+
+                    let mediumIndex;
+                    if (this.selectedFilters && this.selectedFilters.medium && this.selectedFilters.medium[0]) {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/offline/components/library-filters/library-filters.component.ts]: here also use the _.get

**Pull request merged**